plugins/inputs/prometheus/prometheus.go

```diff
@@ -427,6 +433,12 @@ func (p *Prometheus) gatherURL(u URLAndAddress, acc telegraf.Accumulator) error
 	return nil
 }
 
+func (p *Prometheus) addHeaders(req *http.Request) {
+	for header, value := range p.headers {
+		req.Header.Set(header, value)
```

Comment:
The code called req.Header.Add() before but now it calls req.Header.Set(). Any reason for the change? If not, let's switch back to Add to be compatible with the previous code.